File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {


        queue<TreeNode*> q;
        vector<int> ans;

        if(root==nullptr)
            return ans;
        q.push(root);

        while(!q.empty())
        {
            int qz = q.size();
            for(int i=0;i<qz;i++)
            {
                TreeNode *currNode = q.front();
                q.pop();

                if(i==qz-1)
                    ans.push_back(currNode->val);
                
                if(currNode->left!=nullptr)
                    q.push(currNode->left);
                if(currNode->right!=nullptr)
                    q.push(currNode->right);
            }
        }

        return ans;
        
    }
};

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        q=[]
        ans=[]

        if root==None:
            return ans
        
        q.append(root)

        while(len(q)>0):

            qz=len(q)

            for i in range(0,len(q)):
                currNode = q.pop(0)
                if i==qz-1:
                    ans.append(currNode.val)
                if(currNode.left!=None):
                    q.append(currNode.left)
                if(currNode.right!=None):
                    q.append(currNode.right)

        return ans